django/db/backends/utils.py

```diff
@@ -89,10 +93,21 @@ def _execute(self, sql, params, *ignored_wrapper_args):
                 return self.cursor.execute(sql, params)
 
     def _executemany(self, sql, param_list, *ignored_wrapper_args):
+        self._warn_if_apps_not_ready()
         self.db.validate_no_broken_transaction()
         with self.db.wrap_database_errors:
             return self.cursor.executemany(sql, param_list)
 
+    def _warn_if_apps_not_ready(self):
+        if not apps.ready:
+            warnings.warn(
+                "Accessing the database during app initialization is discouraged. "
+                "To silence this warning, avoid executing queries in AppConfig.ready() "
+                "or when your app modules are imported. The Django docs provide "
+                "guidance on how to fix this warning.",
```

Comment:
Can we link to, or name, the exact docs page with info? Otherwise I think we can remove the sentence about the docs, it’s not something we’ve generally done. Searching for the message is usually sufficient.

Reply:
I'd chop `The Django docs ...`

Reply:
Someone named @adamchainz told me that "[we don't usually link to the docs](https://github.com/django/django/pull/16923#discussion_r1218613418)" because those links are hard to keep up-to-date 😉

Reply:
Haha! Well then, let's just drop it as Mariusz says. I clearly can't make my mind up on it.

Reply:
done